### PR TITLE
Feature/write metrics for wat

### PR DESCRIPTION
**Pull request by Brennan1994**

### CODE REVIEW

**Comment by Brennan1994** [comment]
These changes are due to the need to set a specific FPart when writing metrics from a WAT Compute so that models can easily link to the output. To accomplish this, there's a new field with a public setter on dssDatabase that lets us set that fPart override. If that get's set, when writing to the database the record will use that fPart instead of the startTime and Issue Date information it used to. 

That start time and issue date information now gets written to the A Part.

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java [hunk 1]
```diff
@@ -42,6 +42,13 @@ public class DssDatabase implements EnsembleDatabase,MetricDatabase {
     static String metricTimeseriesIdentifier = "MetricTimeseries";
     static String metricPairedDataIdentifier = "MetricPairedData";
     boolean CatalogIsUpToDate = false;
+    private String overriddenFPart = null;
+
+    public void setOverriddenFPart(String overriddenFPart) {
```
Comment (ktarbet) [inline ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java:47]: For this override F part  I have two thoughts to share:
1) consider calling this user-supplied f-part (from the perspective of DssDatabase clients)
2) consider a callback or event or inheritance that actually does the override outside the DssDatabase.

File: ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java [hunk 2]
```diff
@@ -91,7 +98,7 @@ private static String buildDssPath(Ensemble e,int memberNumber, RecordIdentifier
 
     private String buildTimeSeriesStatPathName(RecordIdentifier timeSeriesIdentifier, Duration interval, ZonedDateTime startDateTime, ZonedDateTime issueDate, String stat) {
         DSSPathname path = new DSSPathname();
-        path.setAPart("");
+        path.setAPart(buildTimeAndVersionInfo(startDateTime,issueDate));
```
Comment (ktarbet) [inline ensemble-dss/src/main/java/hec/dss/ensemble/DssDatabase.java:101]: This feels like we are also overriding the A part.  Could buildTimeAndVersionInfo() be called buildApart() to be consistent with the buildFpart().  It looks like the A part and F part are receiving the same information.

**Review by ktarbet — COMMENTED** [review]
My comments are based on (perhaps benefited from) not understanding of the needs of WAT conventions for the F part and A part.

**Comment by ktarbet** [comment]
An specific example of WAT, using setOverriddenFPart(), would be very helpful for my understanding.  Perhaps added to the unit test that uses setOverriddenFPart().

**Pull request closed without merging**